### PR TITLE
Apr 1489 add new assessor applications endpoint

### PR DESCRIPTION
**Pull request by MichaelYoung1981**

### CODE REVIEW

File: src/SFA.DAS.ApplyService.Application.UnitTests/Handlers/AssessorSummaryHandlerTests/AssessorSummaryHandlerTests.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.ApplyService.Application.Apply;
+using SFA.DAS.ApplyService.Application.Apply.Assessor;
+
+namespace SFA.DAS.ApplyService.Application.UnitTests.Handlers.AssessorSummaryHandlerTests
+{
+    class AssessorSummaryHandlerTests
```
Comment (David-B-Read) [inline src/SFA.DAS.ApplyService.Application.UnitTests/Handlers/AssessorSummaryHandlerTests/AssessorSummaryHandlerTests.cs:10]: Missing a TestFixture attribute

File: src/SFA.DAS.ApplyService.Application.UnitTests/Handlers/AssignAssessorHandlerTests/AssignAssessorHandlerTests.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        }
+
+        [Test]
+        public async Task Assigning_application_to_assessor_1_stored_the_assessor_details()
+        {
+            var applicationId = Guid.NewGuid();
+            var expectedUserId = "sadjkffgdji";
+            var expectedUserName = "sadjkffgdji";
+            
+            await _handler.Handle(new AssignAssessorRequest(applicationId, 1, expectedUserId, expectedUserName), new CancellationToken());
```
Comment (David-B-Read) [inline src/SFA.DAS.ApplyService.Application.UnitTests/Handlers/AssignAssessorHandlerTests/AssignAssessorHandlerTests.cs:32]: Using named params would make it easier to see at a glance what 1 and 2 refer to in these tests

File: src/SFA.DAS.ApplyService.Application.UnitTests/Handlers/NewAssessorApplicationsHandlerTests/NewAssessorApplicationsHandlerTests.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+using System.Threading.Tasks;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.ApplyService.Application.Apply;
+using SFA.DAS.ApplyService.Application.Apply.Assessor;
+using SFA.DAS.ApplyService.Domain.Apply;
+
+namespace SFA.DAS.ApplyService.Application.UnitTests.Handlers.NewAssessorApplicationsHandlerTests
+{
+    class NewAssessorApplicationsHandlerTests
```
Comment (David-B-Read) [inline src/SFA.DAS.ApplyService.Application.UnitTests/Handlers/NewAssessorApplicationsHandlerTests/NewAssessorApplicationsHandlerTests.cs:12]: Missing TestFixture

File: src/SFA.DAS.ApplyService.Application/Apply/Assessor/AssessorSummaryHandler.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+            _repository = repository;
+        }
+
+        public async Task<RoatpAssessorSummary> Handle(AssessorSummaryRequest request, CancellationToken cancellationToken)
+        {
+            var newApplicationCount = _repository.GetNewAssessorApplicationsCount(request.UserId);
+
+            await Task.WhenAll(newApplicationCount);
+
+            return new RoatpAssessorSummary(newApplicationCount.Result, 0, 0, 0);
```
Comment (David-B-Read) [inline src/SFA.DAS.ApplyService.Application/Apply/Assessor/AssessorSummaryHandler.cs:23]: Probably stub code for now, but what are the three zeroes?

File: src/SFA.DAS.ApplyService.Data/ApplyRepository.cs [hunk 5]
```diff
@@ -1130,6 +1130,80 @@ public async Task RemoveSectionCompleted(Guid applicationId, Guid applicationSec
...
+                            org.Name AS OrganisationName,
+                            JSON_VALUE(apply.ApplyData, '$.ApplyDetails.UKPRN') AS Ukprn,
+                            JSON_VALUE(apply.ApplyData, '$.ApplyDetails.ReferenceNumber') AS ApplicationReferenceNumber,
+                            JSON_VALUE(apply.ApplyData, '$.ApplyDetails.ProviderRouteName') AS ProviderRoute,
+                            JSON_VALUE(apply.ApplyData, '$.ApplyDetails.ApplicationSubmittedOn') AS SubmittedDate,
+                            Assessor1Name,
+                            Assessor2Name
+	                      FROM Apply apply
+	                      INNER JOIN Organisations org ON org.Id = apply.OrganisationId
+	                      WHERE apply.DeletedAt IS NULL AND apply.GatewayReviewStatus = @gatewayReviewStatusApproved AND (Assessor1UserId IS NULL OR Assessor1UserId <> @userId AND Assessor2UserId IS NULL)
```
Comment (David-B-Read) [inline src/SFA.DAS.ApplyService.Data/ApplyRepository.cs:1150]: I think when we refined the stories, we discussed the fact that someone could assign themselves to Assessor 2 before someone else took on the Assessor 1 role, does this need an extra check?

File: src/SFA.DAS.ApplyService.Database/SFA.DAS.ApplyService.Database.refactorlog [hunk 6]
```diff
@@ -0,0 +1,17 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (David-B-Read) [inline src/SFA.DAS.ApplyService.Database/SFA.DAS.ApplyService.Database.refactorlog:1]: This file should be removed

File: src/SFA.DAS.ApplyService.Database/Tables/Apply.sql [hunk 7]
```diff
@@ -15,7 +15,13 @@
...
     [DeletedAt] DATETIME2 NULL, 
     [DeletedBy] NVARCHAR(256) NULL,
-    [UKPRN] AS JSON_VALUE(ApplyData, '$.ApplyDetails.UKPRN') PERSISTED,
+    [UKPRN] AS JSON_VALUE(ApplyData, '$.ApplyDetails.UKPRN') PERSISTED, 
+    [Assessor1UserId] NVARCHAR(256) NULL, 
+    [Assessor2UserId] NVARCHAR(256) NULL, 
+    [Assessor1Name] NVARCHAR(256) NULL, 
+    [Assessor2Name] NVARCHAR(256) NULL, 
+    [Assessor1ReviewStatus] NVARCHAR(20) NULL, 
+    [Assessor2ReviewStatus] NVARCHAR(20) NULL
```
Comment (David-B-Read) [inline src/SFA.DAS.ApplyService.Database/Tables/Apply.sql:24]: Can the schema change be put into a separate PR so it can be merged into master early

File: src/SFA.DAS.ApplyService.InternalApi.UnitTests/RoatpAssessorControllerTests.cs [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+        {
+            _mediator = new Mock<IMediator>();
+            _controller = new RoatpAssessorController(_mediator.Object);
+        }
+
+        [Test]
+        public async Task Get_summary_returns_summary_for_the_user()
+        {
+            var expectedUser = "sadjkffgdji";
+            var expectedResult = new RoatpAssessorSummary(1, 1, 1, 1);
```
Comment (David-B-Read) [inline src/SFA.DAS.ApplyService.InternalApi.UnitTests/RoatpAssessorControllerTests.cs:31]: Not easy to see at a glance what these numbers are

File: src/SFA.DAS.ApplyService.Database/Tables/Apply.sql (hunk 7 above)
Comment (MichaelYoung1981) [reply src/SFA.DAS.ApplyService.Database/Tables/Apply.sql:24]: I think we're going to need to create a submaster for this anyway as my next piece of work depends on more than just the sql changes.
